File: Implementacion/inteligencia/conMemo/edificio.py

# **************************************************************************
#			  edificio.py  -  description
#   	                      ------------------
# Wizard's Sword
# desarrollado por     : (C) 2004 by M Beatriz Medina Yanez
# *************************************************************************


#  Clase que implementa los atributos de un edificio del juego

# id:       Identificador del edificio

# tipo:     Tipo de edificio
#           0 -> Castillo
#           1 -> Posada
#           2 -> Biblioteca
#           3 -> Tienda de armas
#           4 -> Tienda de escudos
#           5 -> Tienda de armaduras

# llav:     Flag que indica si el edificio esta cerrado o no con llave
#           0 -> No cerrado
#           1 -> Cerrado

# posx:     Posicion x del edificio
# posy:     Posicion y del edificio

# longx:    Longitud x del edificio
# longy:    Longitud y del edificio

# casilx:   Numero de casillas que ocupa en el eje x
# casily:   Numero de casillas que ocupa en el eje y
       
# nAtr:     Numero de atributos del objeto Edificio

class Edificio:
    __id = None      
    __tipo = None   
    __llav = None   
    __posx = None   
    __posy = None
    __longx = None
    __longy = None
    __casilx = None
    __casily = None
    __nAtr = None

    # Inicializador
    def __init__ (self):
        self.__nAtr = 9

    # Metodo consultor que devuelve el numero de atributos del objeto Edificio
    def obtieneNAtributos (self):
        return self.__nAtr
    
    # Metodo consultor que devuelve el valor del atributo especificado
    def obtieneAtributo (self, atributo):

        if ((atributo == 1) or (atributo == 'id')):
            return self.__id

        elif ((atributo == 2) or (atributo == 'tipo')):
            return self.__tipo

        elif ((atributo == 3) or (atributo == 'llav')):
            return self.__llav

        elif ((atributo == 4) or (atributo == 'posx')):
            return self.__posx

        elif ((atributo == 5) or (atributo == 'posy')):
            return self.__posy

        elif ((atributo == 6) or (atributo == 'longx')):
            return self.__longx

        elif ((atributo == 7) or (atributo == 'longy')):
            return self.__longy

        elif ((atributo == 8) or (atributo == 'casilx')):
            return self.__casilx

        elif ((atributo == 9) or (atributo == 'casily')):
            return self.__casily

    # Metodo que actualiza con un nuevo valor al atributo especificado    
    def reAsignaAtributo (self, atributo, nuevoValor):

        if ((atributo == 1) or (atributo == 'id')):
            self.__id = nuevoValor

        elif ((atributo == 2) or (atributo == 'tipo')):
            self.__tipo = nuevoValor
            
        elif ((atributo == 3) or (atributo == 'llav')):
            self.__llav = nuevoValor

        elif ((atributo == 4) or (atributo == 'posx')):
            self.__posx = nuevoValor

        elif ((atributo == 5) or (atributo == 'posy')):
            self.__posy = nuevoValor

        elif ((atributo == 6) or (atributo == 'longx')):
            self.__longx = nuevoValor

        elif ((atributo == 7) or (atributo == 'longy')):
            self.__longy = nuevoValor

        elif ((atributo == 8) or (atributo == 'casilx')):
            self.__casilx = nuevoValor

        elif ((atributo == 9) or (atributo == 'casily')):
            self.__casily = nuevoValor

# *************************************************************************

File: Implementacion/edificio.cc

/***************************************************************************
                          edificio.cc  -  description
                             -------------------
  Wizard's Sword
  desarrollado por     : (C) 2004 by Mª Beatriz Medina Yañez
 ***************************************************************************/

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#include "edificio.hh"

/***************************************************************************/

/* Constructor del objeto Edificio que inicializa todos sus atributos a 
   VACIO */
Edificio::Edificio ()
{
    // Inicialización de los atributos a VACIO
    this->id = VACIO;
    this->tipo = VACIO;
    this->llav = VACIO;
    this->posx = VACIO;
    this->posy = VACIO;
    this->longx = VACIO;
    this->longy = VACIO;
    this->casilx = VACIO;
    this->casily = VACIO;

    // Inicialización del número de atributos
    this->nAtr = 9;
}

/***************************************************************************/

// Destructor del objeto Edificio
Edificio::~Edificio ()
{}

/***************************************************************************/

// Método consultor que devuelve el número de atributos del objeto Edificio
int Edificio::obtieneNAtributos (void)
{
    return this->nAtr;
}

/***************************************************************************/

/* Método consultor que devuelve el valor del atributo cuyo nombre se 
   especifica */ 
int Edificio::obtieneAtributo (char *atributo)
{
    if (!strcmp (atributo, "id"))
	return this->id;

    else if (!strcmp (atributo, "tipo"))
	return this->tipo;

    else if (!strcmp (atributo, "llav"))
	return this->llav;

    else if (!strcmp (atributo, "posx"))
	return this->posx;

    else if (!strcmp (atributo, "posy"))
	return this->posy;

    else if (!strcmp(atributo, "longx"))
	return this->longx;

    else if (!strcmp(atributo, "longy"))
	return this->longy;

    else if (!strcmp(atributo, "casilx"))
	return this->casilx;

    else if (!strcmp(atributo, "casily"))
	return this->casily;

    else
    {
	fprintf (stderr, "edificio.cc::obtieneAtributo (char *): ");
	fprintf (stderr, "El atributo %s no existe\n", atributo);
	exit (-1);
    }
}

/***************************************************************************/

/* Método consultor que devuelve el valor del atributo cuyo índice se 
   especifica */ 
int Edificio::obtieneAtributo (int atributo)
{
    if (atributo == 1)
	return this->id;

    else if (atributo == 2)
	return this->tipo;

    else if (atributo == 3)
	return this->llav;

    else if (atributo == 4)
	return this->posx;

    else if (atributo == 5)
	return this->posy;

    else if (atributo == 6)
	return this->longx;

    else if (atributo == 7)
	return this->longy;

    else if (atributo == 8)
	return this->casilx;

    else if (atributo == 9)
	return this->casily;

    else
    {
	fprintf (stderr, "edificio.cc::obtieneAtributo (char *): ");
	fprintf (stderr, "El atributo %s no existe\n", atributo);
	exit (-1);
    }
}

/***************************************************************************/

/* Método que actualiza con un nuevo valor al atributo cuyo nombre se
   especifica */
void Edificio::reAsignaAtributo (char *atributo, int nuevoValor)
{
    if (!strcmp (atributo, "id"))
	this->id = nuevoValor;

    else if (!strcmp (atributo, "tipo"))
	this->tipo = nuevoValor;

    else if (!strcmp (atributo, "llav"))
	this->llav = nuevoValor;

    else if (!strcmp (atributo, "posx"))
	this->posx = nuevoValor;

    else if (!strcmp (atributo, "posy"))
	this->posy = nuevoValor;

    else if (!strcmp(atributo, "longx"))
	this->longx = nuevoValor;

    else if (!strcmp(atributo, "longy"))
	this->longy = nuevoValor;

    else if (!strcmp(atributo, "casilx"))
	this->casilx = nuevoValor;

    else if (!strcmp(atributo, "casily"))
	this->casily = nuevoValor;

    else
    {
	fprintf (stderr, "edificio.cc::reAsignaAtributo (char *, int): ");
	fprintf (stderr, "El atributo %s no existe\n", atributo);
	exit (-1);
    }    
}

/***************************************************************************/

/* Método que actualiza con un nuevo valor al atributo cuyo índice se
   especifica */
void Edificio::reAsignaAtributo (int atributo, int nuevoValor)
{
    if (atributo == 1)
	this->id = nuevoValor;

    else if (atributo == 2)
	this->tipo = nuevoValor;

    else if (atributo == 3)
	this->llav = nuevoValor;

    else if (atributo == 4)
	this->posx = nuevoValor;

    else if (atributo == 5)
	this->posy = nuevoValor;

    else if (atributo == 6)
	this->longx = nuevoValor;

    else if (atributo == 7)
	this->longy = nuevoValor;

    else if (atributo == 8)
	this->casilx = nuevoValor;

    else if (atributo == 9)
	this->casily = nuevoValor;

    else
    {
	fprintf (stderr, "edificio.cc::reAsignaAtributo (char *, int): ");
	fprintf (stderr, "El atributo %s no existe\n", atributo);
	exit (-1);
    }    
}

/***************************************************************************/
